classes/Pods.php

```diff
@@ -1103,51 +1036,113 @@ public function field( $name, $single = null, $raw = false ) {
 
 					$object_field_found = true;
 				}
-			} elseif ( 0 === strpos( $params->name, 'image_attachment.' ) ) {
-				$size = 'thumbnail';
+			}
 
-				$image_id = 0;
+			if ( ! $object_field_found ) {
+				foreach ( $image_fields as $image_field ) {
+					if ( isset( $this->fields[ $image_field ] ) ) {
+						// Skip any names that are registered fields within this Pod.
+						continue;
+					}
+					if (
+						$image_field === $params->name ||
+						0 === strpos( $params->name, $image_field . '.' )
+					) {
+						// Is it a URL request?
+						$url = '_url' === substr( $image_field, -4 );
+						if ( $url ) {
+							$image_field = substr( $image_field, 0, -4 );
+						}
 
-				$field_names = explode( '.', $params->name );
+						// Results in an empty array if no traversal names are passed.
+						$traverse_names = explode( '.', $params->name );
+						array_shift( $traverse_names );
+
+						$attachment_id = 0;
+						switch ( $image_field ) {
+							case 'post_thumbnail':
+								$attachment_id = get_post_thumbnail_id();
+								break;
+							case 'image_attachment':
+								if ( isset( $traverse_names[0] ) ) {
+									$attachment_id = $traverse_names[0];
+									array_shift( $traverse_names );
+								}
+								break;
+						}
 
-				if ( isset( $field_names[1] ) ) {
-					$image_id = $field_names[1];
-				}
+						if ( $attachment_id ) {
+
+							$size = 'thumbnail';
+							if ( isset( $traverse_names[0] ) ) {
+								$size  = $traverse_names[0];
+								$sizes = get_intermediate_image_sizes();
```

Comment:
:no_entry_sign: Intermediate images do not exist on the VIP platform, and thus get_intermediate_image_sizes() returns an empty array() on the platform. This behavior is intentional to prevent WordPress from generating multiple thumbnails when images are uploaded.

<sub>hash: 7c8eea3bea2dc4d02551359ca50b07af</sub>

Reply:
@sc0ttkclark Thoughts?